imod_coupler/drivers/dfm_metamod/dfm_metamod.py

```diff
@@ -655,21 +668,21 @@ def exchange_correction_dflow2mf6(self, qdfm_realised: NDArray[float_]) -> None:
         the drainage/inflitration flux to the 1d rivers as realised by DFM is passed to
         mf6 as a correction
         """
-
-        # mf6 riv-active demand, dims=mf6
-        qmf6_demand = self.mf6.get_river_flux_estimate(
-            self.coupling.mf6_model, self.coupling.mf6_river_active_pkg
-        )
-        qmf6_demand_sec = -qmf6_demand / days_to_seconds(self.delt_mf6)
-
-        # mf6 riv-active demand, dims=dfm
-        qdfm_demand = self.exchange_balans_1d.demand["mf-riv2dflow1d_flux"]
-
-        qmf_corr = self.mapping.calc_correction(
-            self.map_active_mod_dflow1d["mf-riv2dflow1d_flux"],
-            qmf6_demand_sec,
-            qdfm_demand,
-            qdfm_realised,
+        wbal = self.exchange_balans_1d
+        realised_dfm = wbal.realised["dflow1d_flux2mf-riv_negative"]
+        demand_pos = wbal.demand["mf-riv2dflow1d_flux_positive"]
+        demand_neg = wbal.demand["mf-riv2dflow1d_flux_negative"]
+        mask = np.less(0.0, demand_neg)
+        realised_fraction = realised_dfm * 0.0 + 1.0
+        realised_fraction[mask] = (realised_dfm[mask] - demand_pos[mask]) / demand_neg[
```

Comment:
do we need a sanity check to see if the realised_fraction < 0 (for cases where realised > demand). This should not occur...